packages/features/data-table/components/filters/DateRangeFilter.tsx

```diff
@@ -152,72 +165,91 @@ export const DateRangeFilter = ({ column, showClearButton = false }: DateRangeFi
 
   const isCustomPreset = selectedPreset.value === CUSTOM_PRESET_VALUE;
 
+  let customButtonLabel = t("date_range");
+  if (startDate && endDate) {
+    customButtonLabel = `${format(startDate.toDate(), "LLL dd, y")} - ${format(
+      endDate.toDate(),
+      "LLL dd, y"
+    )}`;
+  } else if (startDate) {
+    customButtonLabel = `${format(startDate.toDate(), "LLL dd, y")} - ?`;
+  }
+
   return (
     <Popover>
       <PopoverTrigger asChild>
-        <Button color="secondary" className="items-center capitalize" EndIcon="chevron-down">
+        <Button
+          color="secondary"
+          className="items-center capitalize"
+          StartIcon="calendar-range"
+          EndIcon="chevron-down">
           {!isCustomPreset && <span>{t(selectedPreset.labelKey, selectedPreset.i18nOptions)}</span>}
-          {isCustomPreset &&
-            (endDate ? (
-              <span>
-                {format(startDate.toDate(), "LLL dd, y")} - {format(endDate.toDate(), "LLL dd, y")}
-              </span>
-            ) : (
-              <span>{format(startDate.toDate(), "LLL dd, y")} - End</span>
-            ))}
+          {isCustomPreset && <span>{customButtonLabel}</span>}
         </Button>
       </PopoverTrigger>
       <PopoverContent className="flex w-fit p-0" align="end">
         {isCustomPreset && (
           <div className="border-subtle border-r">
             <DateRangePicker
               dates={{
-                startDate: startDate.toDate(),
+                startDate: startDate?.toDate(),
                 endDate: endDate?.toDate(),
               }}
-              minDate={currentDate.subtract(2, "year").toDate()}
-              maxDate={currentDate.toDate()}
+              minDate={forcePast ? currentDate.subtract(2, "year").toDate() : null}
+              maxDate={forcePast ? currentDate.toDate() : undefined}
               disabled={false}
               onDatesChange={updateDateRangeFromPicker}
               withoutPopover={true}
             />
+            {forceCustom && (
```

Comment:
we need an extra "clear" button below the calendar view in case of "forceCustom" because we don't have this view when `forceCustom`:

![Screenshot 2025-02-25 at 13 36 40](https://github.com/user-attachments/assets/8fe35d7e-6515-45e3-a446-3dfc8f8206e5)